ServiceWebsite/ServiceWebsite.AcceptanceTests/Hooks/DataSetUp.cs

```diff
@@ -1,94 +1,64 @@
 using ServiceWebsite.AcceptanceTests.Contexts;
-using ServiceWebsite.AcceptanceTests.Models;
-using ServiceWebsite.BookingsAPI.Client;
-using ServiceWebsite.Common;
-using ServiceWebsite.Configuration;
-using System.Collections.Generic;
-using System.Linq;
-using System.Net;
+using ServiceWebsite.AcceptanceTests.Helpers;
+using ServiceWebsite.AcceptanceTests.TestClients;
+using ServiceWebsite.AcceptanceTests.Validations;
+using System.Reflection;
 using TechTalk.SpecFlow;
 
 namespace ServiceWebsite.AcceptanceTests.Hooks
 {
     [Binding]
     public class DataSetUp
     {
-        [BeforeScenario(Order = 0)]
-        public void OneTimeSetup(TestContext testContext)
-        {
-            var configRootBuilder = new ConfigurationBuilder()
-             .AddJsonFile("appsettings.json")
-             .AddEnvironmentVariables()
-             .AddUserSecrets("CF5CDD5E-FD74-4EDE-8765-2F899C252122");
-            var configRoot = configRootBuilder.Build();
+        private static BookingsApiClientHelper _bookingsApiHelper;
+        private static DataSetUpValidation _dataSetUpValidation;
+        private static IConfigurationRoot _configRoot;
+        private static BookingsApiTestClient _client;
+        public TestContext TestContext { get; set; }
 
-            var azureAdConfig = Options.Create(configRoot.GetSection("AzureAd").Get<SecuritySettings>()).Value;
-            var vhServiceConfig = Options.Create(configRoot.GetSection("VhServices").Get<ServiceSettings>()).Value;
-            var userAccountConfig = Options.Create(configRoot.GetSection("TestUserSecrets").Get<UserAccount>()).Value;
-            var authContext = new AuthenticationContext(azureAdConfig.Authority);
-            var credential = new ClientCredential(azureAdConfig.ClientId, azureAdConfig.ClientSecret);
-            testContext.BearerToken = authContext.AcquireTokenAsync(vhServiceConfig.BookingsApiResourceId, credential).Result.AccessToken;
-            testContext.BaseUrl = vhServiceConfig.BookingsApiUrl;
-            testContext.TestUserSecrets = userAccountConfig;
-            testContext.AzureAd = azureAdConfig;
-            testContext.WebsiteUrl = configRoot.GetSection("WebsiteUrl").Value;
-            testContext.VideoAppUrl = configRoot.GetSection("VideoAppUrl").Value;
+        public DataSetUp()
+        {
+            _bookingsApiHelper = new BookingsApiClientHelper();
+            _dataSetUpValidation = new DataSetUpValidation();
         }
 
-        [BeforeScenario(Order = 2)]
-        public static void ClearAnyHearings(TestContext context)
+        [BeforeScenario(Order = 0)]
+        public void OneTimeSetup(TestContext testContext)
         {
-            ClearHearings(context, context.TestUserSecrets.Individual);
-            ClearHearings(context, context.TestUserSecrets.Representative);
+            _configRoot = ConfigurationHelper.BuildDefaultConfigRoot();
+            _client = _bookingsApiHelper.CreateClientWithConfig(_configRoot);
```

Comment:
Dont think you need to pass _configRoot everywhere, you have the TestContext, just add what you need to it.

Reply:
Also Specflow has Dependency Injection, meaning you can pass the TestContext in the contructor of any class.
https://github.com/techtalk/SpecFlow/wiki/Context-Injection

Reply:
done